spec/services/get_product_service_spec.rb

```diff
@@ -18,6 +18,10 @@
     context 'when product_id is nil' do
       let(:product_id) { nil }
 
+      before do
+        allow(CacheManager).to receive(:fetch_value).and_return(nil)
```

Comment:
this validate will works on both cache and db level, if test only for cache, what happens for db level, => don't need to specific the validation from cache level, this should be a logic that not depend on either cache or db level => remove